src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -861,12 +861,14 @@ export class FolderMatchWorkspaceRoot extends FolderMatchWithResource {
 		const fileMatchParentParts: URI[] = [];
 		let uri = this.uriParent(rawFileMatch.resource);
 
+		let prev_uri;
 		while (!this.uriEquals(this.resource, uri)) {
-			if (uri.path === '/') {
-				throw Error(`${rawFileMatch.resource} is not correctly configured as a child of its ${this.resource}`);
-			}
 			fileMatchParentParts.unshift(uri);
+			prev_uri = uri;
 			uri = this.uriParent(uri);
+			if (this.uriEquals(prev_uri, uri)) {
```

Comment:
Is this checking when you get to the root or is there another case where it will loop?

Reply:
Yeah, it's checking for when you get to the root so that it doesn't loop forever. This should be the only case that would potentially cause an infinite loop, since the .dirname() will only return the same as the input or higher in the path structure.